pkg/ccl/changefeedccl/authorization.go

```diff
@@ -18,15 +18,24 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgnotice"
 	"github.com/cockroachdb/cockroach/pkg/sql/privilege"
 	"github.com/cockroachdb/cockroach/pkg/sql/roleoption"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/syntheticprivilege"
 	"github.com/cockroachdb/errors"
 )
 
 func checkPrivilegesForDescriptor(
 	ctx context.Context, p sql.PlanHookState, desc catalog.Descriptor,
 ) (hasSelect bool, hasChangefeed bool, err error) {
-	if desc.GetObjectType() != privilege.Table {
-		return false, false, errors.AssertionFailedf("expected descriptor %d to be a table descriptor, found: %s ", desc.GetID(), desc.GetObjectType())
+	if desc == nil {
+		return false, false, errors.AssertionFailedf("expected descriptor to be non-nil")
+	}
+	if objectType := desc.GetObjectType(); objectType != privilege.Table &&
+		objectType != privilege.Database {
```

Comment:
nit: might be cleaner as a switch statement with an empty block for the matching types case and returning the error in the default case